thanos/charts/templates/ruler/ruler-svc.yaml

```diff
@@ -0,0 +1,30 @@
+# SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Greenhouse contributors
+# SPDX-License-Identifier: Apache-2.0
+
+apiVersion: v1
+kind: Service
+metadata:
+  {{- if .Values.thanos.ruler.annotations }}
```

Comment:
There is no way to pass labels through the ThanosRuler CR?

Reply:
does not work, checking with prometheus-operator maintainers

Reply:
@richardtief Here is the link to the Upstream GitHub issue created by @trouaux - https://github.com/prometheus-operator/prometheus-operator/issues/7317

Reply:
>btw if you're not already aware, it's also possible to configure the operator to add labels & annotations to all managed objects via the --labels and --annotations CLI args (depending on your use case, it may or may not be useful).

We can also do this as a workaround?

Reply:
just gave this a try again and it doesn't work for me. I'll double check again with maintainers. Maybe we have a second issue

Reply:
Ok, just figured out that prometheus-operator does not deploy a service to access the UI of the Thanos Rule. It's just a `operated` - headless service for the operator to push changes agains the Thanos Rule API. So we must create our own.

Reply:
There is also thanos-rule charts available in the [kube-prometheus-stack](https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-prometheus-stack/templates/thanos-ruler), which also includes a dedicated service. E
Example: https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-prometheus-stack/templates/thanos-ruler